### ISSUE TITLE
Generator<T> can conflict with TypeScript built-in Generator

### ISSUE THREAD

**Issue by richardboydnz** [bctnry/gemtext#1]
In case you are interested:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-6.html
```
interface Generator<T = unknown, TReturn = any, TNext = unknown>
  extends Iterator<T, TReturn, TNext> {
  next(...args: [] | [TNext]): IteratorResult<T, TReturn>;
  return(value: TReturn): IteratorResult<T, TReturn>;
  throw(e: any): IteratorResult<T, TReturn>;
  [Symbol.iterator](): Generator<T, TReturn, TNext>;
}
```

If you want to use generators to stream HTML output (though not really important for the size of a typical  GemText file), your Generator<T> definition would conflict with the TypeScript generators.

I'm only looking at this repository for research, so don't change it for my sake - just letting you know about something you would have to fix if you want to promote this library.

**Comment by bctnry** [comment]
thanks for pointing out. would fix.

**Comment by bctnry** [comment]
should be fixed with 47ca6a2.
